src/test/java/org/apache/commons/validator/routines/DomainValidatorStartupTest.java

```diff
@@ -35,24 +39,52 @@
 @RunWith( ClassLoaderPerTestRunner.class )
 public class DomainValidatorStartupTest {
 
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void testUpdateBaseArrayCC() {
-        DomainValidator.updateTLDOverride(ArrayType.COUNTRY_CODE_RO, new String[]{"com"});
+        // FIXME Simplification once upgraded to Java 1.8
+        final ThrowingRunnable testMethod = new ThrowingRunnable() {
+            public void run() {
+                DomainValidator.updateTLDOverride(ArrayType.COUNTRY_CODE_RO, new String[]{"com"});
+            }
+        };
+        final IllegalArgumentException thrown = assertThrows(IllegalArgumentException.class, testMethod);
+        assertThat(thrown.getMessage(), is(equalTo("Cannot update the table: COUNTRY_CODE_RO")));
```

Comment:
Same comment as in the Apache Commons Collections PR. Let's skip asserting the exception messages.

Reply:
Why skip? It's a common conversation I always keep having... I just wondered your rational...

My argument to alway checking the exception message is how do you know the test is failing as you are expecting it to fail?

You write a test to check X, and it throws Y exception. If Z might also causes Y to be thrown, so how do you know your testing X? My view is if you skip checking the exception message then you might as well just delete the test.

So Digester BeanPropertySetterRuleTestCase.testSetUnknownProperty according to the error messages and comments should produce an NoSuchMethodException, but currently is throws org.xml.sax.SAXException due to a code path meaning any non expected Exception is ignored.

So this test is to check COUNTRY_CODE_RO, but if the code behind the test changes and IllegalArgumentException is still thrown, how do you know it's the correct exception for what you have setup?